### DISCUSSION TITLE
Is there any way to get the root node of all application resource trees in a given project?

### DISCUSSION THREAD

**Discussion by xtdong1001** [argoproj/argo-cd#7240] [Q&A]
Hello, I used /api/v1/applications?project={projectName} to get all applications of a project. The thing is there are 4000+ applications in our environment and it takes 4+ seconds to answer the request.

So I was wondering if I can use any query parameters to get just the root node of all application resource trees in a given project?

**Comment by wanghong230** [comment]
I believe the attribute selection is supported which could be helpful.

https://HostName/api/v1/applications?project=default&&fields=metadata.resourceVersion%2Citems.metadata.name%2Citems.metadata.annotations%2Citems.metadata.labels%2Citems.metadata.creationTimestamp%2Citems.metadata.deletionTimestamp%2Citems.spec%2Citems.operation.sync%2Citems.status.sync.status%2Citems.status.health%2Citems.status.operationState.phase%2Citems.status.operationState.operation.sync%2Citems.status.summary&selector=

**Reply by wanghong230** [reply]
You can use the ArgoCD UI to check the developer console network to mimic the API request.

**Comment by wanghong230** [comment]
I talked with the team. The application is cached in memory, so the slowness could be from simply the data transfer or from the extra logic execution time due to RBAC enforcement. From the developer console for a particular request, you may be able to see the latency breakdown to have more clues.

I recommend doing such an experiment. Could you use the root admin account to the same query to see whether it will be helpful to the performance? The RBAC/Casbin for an SSO user could bear some additional processing time. In additional. Then, use the field selector could reduce the bytes transferred.
